### ISSUE TITLE
GoPro HERO5 Black raw "format"

### ISSUE THREAD

**Issue by LebedevRI** [darktable-org/rawspeed#40]
Just so it does not get lost
Copied from https://redmine.darktable.org/issues/11466

```
The .GPR format is an extension Adobe’s .DNG format, adding enhanced compression (VC5 Standard)

A sample file:
https://raw.pixls.us/getfile.php/936/raw/GOPR0020.GPR

FAQ from GoPro about the file format:
https://gopro.com/help/articles/Question_Answer/What-is-RAW-Format

An interesting blogpost about the "VC5 Standard" 
https://medium.com/@kierank_/reverse-engineering-the-gopro-cineform-codec-7411312bfe1c#.9f7ztpjo8
```

**Comment by supertobi** [comment]
Here is the jpeg version of the photo:

![gopr0020](https://cloud.githubusercontent.com/assets/653976/22122136/0f306544-de87-11e6-8858-190214f5f714.JPG)

**Comment by supertobi** [comment]
GoPro HERO5 Black RawTherapee issue:
https://github.com/Beep6581/RawTherapee/issues/3569

**Comment by shoffmeister** [comment]
https://trac.ffmpeg.org/ticket/1087 

https://git.ffmpeg.org/gitweb/ffmpeg.git/blob_plain/HEAD:/libavcodec/cfhd.c <-- that is supposed to decode the video, so the compression might serve as a template? 

This codec goes by VC-5 (misnamed, it seems), cfhd, CineForm HD.

Also: http://cineform.com/compression-subsystem

**Comment by supertobi** [comment]
GoPro Open Sources the CineForm Codec:
https://gopro.com/news/gopro-open-sources-the-cineform-codec

https://github.com/gopro/cineform-sdk

**Comment by LebedevRI** [comment]
Yep, seen that yesterday.
I suppose libraw will want to link to that.
As for rawspeed, i suppose a clean, minimal implementation (with less horrible code than that, and no double/triple-licensing shenanigans) could be added by implementing from scratch.

**Comment by LibRaw** [comment]
>> I suppose libraw will want to link to that.

Most likely no in (coming) 0.19, but most likely yes in next version cycle

**Comment by pzemlyan** [comment]
General Purpose Raw image format API was released: https://github.com/gopro/gpr
Also a few HERO6 samples was uploaded: https://raw.pixls.us/getfile.php/2556/nice/GoPro%20-%20HERO6%20Black%20-%2016bit%20(4:3).GPR

**Comment by supertobi** [comment]
GoPro just released the HERO7, are there any news here ore in @LibRaw?

**Comment by LebedevRI** [comment]
Here - no, nothing has changed.

After reading through the existing crumbs of info on the Canon's CRX codec,
they both seem to be based on wavelets, which is, i suppose, not all that surprising.
So _maybe_ after Canon's CRX codec is implemented,
_maybe_ this could be implemented on top of that code.

**Comment by lclevy** [comment]
https://github.com/lclevy/canon_cr3/blob/master/readme.md#crx-compression

**Comment by stloeffler** [comment]
As I needed this myself, I implemented a basic GPR decompressor in #146 - I hope this helps.

**Comment by supertobi** [comment]
Btw. I just discovered this project:
https://github.com/emericg/libcineform
